server/src/main/java/org/opensearch/cluster/health/ClusterShardHealth.java

```diff
@@ -229,9 +230,36 @@ public void writeTo(final StreamOutput out) throws IOException {
      *     Shard health is YELLOW when primary shard is active but at-least one replica shard is inactive.
      *     Shard health is RED when the primary is not active.
      * </p>
+     * <p>
+     *     In search-only mode (when {@link IndexMetadata#INDEX_BLOCKS_SEARCH_ONLY_SETTING} is enabled):
+     * </p>
+     * <ul>
+     *     <li>Shard health is GREEN when all expected search replicas are active</li>
+     *     <li>Shard health is YELLOW when some (but not all) search replicas are active</li>
+     *     <li>Shard health is RED when no search replicas are active</li>
+     * </ul>
      */
-    public static ClusterHealthStatus getShardHealth(final ShardRouting primaryRouting, final int activeShards, final int totalShards) {
-        assert primaryRouting != null : "Primary shard routing can't be null";
+    public static ClusterHealthStatus getShardHealth(
+        final ShardRouting primaryRouting,
+        final int activeShards,
+        final int totalShards,
+        final IndexMetadata indexMetadata
+    ) {
+        boolean isSearchOnlyEnabled = indexMetadata.getSettings()
```

Comment:
pls inline this after checking for null routing.

Reply:
I have updated this in my latest commit, please check.